pennylane/templates/subroutines/uccsd.py

```diff
@@ -213,12 +179,24 @@ def circuit(weights, wires, ph=ph, pphh=pphh, init_state=ref_state):
         ),
     )
 
+    for pphh_ in pphh:
+        if len(pphh_) != 2:
+            raise ValueError(
+                "expected entries of pphh to be of size 2; got {} of length {}".format(
+                    pphh_, len(pphh_)
+                )
+            )
+
     ###############
 
     qml.BasisState(np.flip(init_state), wires=wires)
 
-    for d, i_pphh in enumerate(pphh):
-        DoubleExcitationUnitary(weights[len(ph) + d], wires=i_pphh)
+    # turn wire arguments into Wires objects
+    ph = [Wires(w) for w in ph]
+    pphh = [[Wires(w1), Wires(w2)] for w1, w2 in pphh]
```

Comment:
That should be easy enough to do. I'm sorry if the qchem pipeline gets one more step, but this may become quite crucial when the devices start getting interesting...

What about these functions?

```python

def ph_wires(ph_locations, wires=None):
    """XXX
    
    Args:
        ph_locations (list[list[int]]): list of indices of the two qubits representing a particle-hole pair
        wires (Iterable[Any]): Wires of the quantum device. If None, will use consecutive wires.
    
    Returns:
         list[list[Any]]
    """

    if wires is None:
        wires = range(np.max(ph_locations)+1)

    ph = []
    for p, h in ph_locations:
        ph_wires = [wires[i] for i in range(p, h + 1)]
        ph.append(ph_wires)
    return ph


def pphh_wires(pphh_locations, wires=None):
    """XXX

    Args:
        pphh_locations (list[list[int]]): list of indices of the four qubits representing two particle-hole pairs
        wires (Iterable[Any]): Wires of the quantum device. If None, will use consecutive wires.

    Returns:
         list[list[list[Any]]]
    """
    if wires is None:
        wires = range(np.max(ph_locations)+1)

    pphh = []

    for p1, p2, h1, h2  in pphh_locations:
        pphh1_wires = [wires[i] for i in range(p1, p2 + 1)]
        pphh2_wires = [wires[i] for i in range(h1, h2 + 1)]
        pphh.append([pphh1_wires, pphh2_wires])

    return pphh

# output of sd_excitations:
ph_locations = [[0, 2], [0, 4]]
pphh_locations = [[0, 1, 2, 4], [0, 1, 3, 4]]

# wires the user chooses
wires = ['q0', 'q1', 'q2', 'q3', 'q4']

print(ph_wires(ph_locations, wires))
# [['q0', 'q1', 'q2'], ['q0', 'q1', 'q2', 'q3', 'q4']]

print(pphh_wires(pphh_locations, wires))
# [[['q0', 'q1'], ['q2', 'q3', 'q4']], [['q0', 'q1'], ['q3', 'q4']]]

print(ph_wires(ph_locations))
# [[0, 1, 2], [0, 1, 2, 3, 4]]

print(pphh_wires(pphh_locations))
# [[[0, 1], [2, 3, 4]], [[0, 1], [3, 4]]]

```

Reply:
Would you mind implementing them in the qchem module (separately from this PR)?
You know the workflow best...

Reply:
Exactly. Thanks @mariaschuld, this is very nice. I can implement this in the ``qchem`` module.